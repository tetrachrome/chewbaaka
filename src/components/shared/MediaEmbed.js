/**
 * MediaEmbed.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 09, 2020
 * Updated  : Aug 09, 2020
 */

import React from 'react'

export default function MediaEmbed({ title, width, height, src }) {
  width = width ? width : 560;
  height = height ? height : 315;

  title = title ? title : "YouTube video";

  const componentDimensionStyle = {
    width: width,
    height: height
  };

  return (
    <div style={componentDimensionStyle}>
      <iframe
        title={title}
        width={width}
        height={height}
        src={src}
        frameBorder="0"
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen>
      </iframe>
    </div>
  );
}
