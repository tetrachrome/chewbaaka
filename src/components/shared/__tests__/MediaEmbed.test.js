/**
 * MediaEmbed.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 17, 2020
 * Updated  : Aug 17, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import MediaEmbed from '../MediaEmbed'

const title = "This is a title";
const width = 480;
const height = 320;
const src = "https://www.youtube.com/watch?v=IRAbmfnrkL8";

test('renders MediaEmbed component', () => {
  render(
    <MediaEmbed
      title={title}
      width={width}
      height={height}
      src={src}
    />
  );
});

test('MediaEmbed component snapshot', () => {
  const tree = renderer
    .create(
      <MediaEmbed
        title={title}
        width={width}
        height={height}
        src={src}
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
