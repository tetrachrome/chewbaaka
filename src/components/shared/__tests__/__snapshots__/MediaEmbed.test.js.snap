// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaEmbed component snapshot 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowFullScreen={true}
  className="FullWidth"
  frameBorder="0"
  height={320}
  src="https://www.youtube.com/watch?v=IRAbmfnrkL8"
  title="This is a title"
  width={480}
/>
`;
